### PR TITLE
feature/Ny modul Modal CU 865cn09xb

### PR DESCRIPTION
**Pull request by annalinneajohansson**
- (add) ACF fields for module Modal
- (add) New module "Modal"

### CODE REVIEW

**Comment by helsingborgsstad** [comment]
Task linked: [CU-865cn09xb Ny modul: Modal](https://app.clickup.com/t/865cn09xb)

File: source/php/Module/Modal/Modal.php [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+    /**
+     * Disallow the Modal block for the current post type;
+     * modals inside of modals can possibly create infinite loops.
+     *
+     * @param array $allowedBlockTypes The allowed block types.
+     * @param object $editorContext The editor context.
+     * @return array The updated allowed block types.
+     */
+    public function disallowBlockType($allowedBlockTypes, $editorContext)
```
Comment (sebastianthulin) [inline source/php/Module/Modal/Modal.php:98]: Can be replaced with this option on module creation. 
https://github.com/helsingborg-stad/Modularity/blob/3.0/develop/source/php/Module.php#L105

File: source/php/Module/Modal/Modal.php [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        $this->postType = 'modal-content';
+
+        add_action('init', [$this, 'registerPostType'], 99);
+        add_filter('allowed_block_types_all', array($this, 'disallowBlockType'), 10, 2);
+    }
+
+    public function data(): array
+    {
+
+        $fields = get_fields($this->ID);
```
Comment (sebastianthulin) [inline source/php/Module/Modal/Modal.php:26]: Use https://github.com/helsingborg-stad/Modularity/blob/3.0/develop/source/php/Module.php#L284 instead.

**Review by sebastianthulin — CHANGES_REQUESTED** [review]
Small changes, otherwise great!

File: source/php/Module/Modal/Modal.php (hunk 1 above)
Comment (annalinneajohansson) [reply source/php/Module/Modal/Modal.php:98]: I'm not following your thought here; can you please explain?

**Pull request merged**